### ISSUE TITLE
remote functions for private github repo

### ISSUE THREAD

**Issue by cderv** [hadley/devtools#1403]
Hi, 

I tried to used `Remotes` field in `DESCRIPTION` to manage dependencies for packages of my company hosted a private organisation.

I do not manage yet to make it works and I have notice some problems regarding private vs public repo. 

``` r
# with public repo
devtools:::remote_package_name.github_remote(devtools:::github_remote("hadley/lubridate"))
#> Using GitHub PAT from envvar GITHUB_PAT
#> [1] "lubridate"

# with private repo
devtools:::remote_package_name.github_remote(devtools:::github_remote("cderv/myprivaterepo"))
#> Using GitHub PAT from envvar GITHUB_PAT
#> [1] NA
```

In fact, the `GET` request in `remote_package_name.github_remote` has a result of an error 404. 
As there is no authentication sent with the `GET` using `devtools::github_auth(github_pat())`, it cannot found the page.

I tried first as it is done inside `remote_package_name.github_remote`
```r
httr::GET("https://github.com", 
                path = "cderv/myprivaterepo/raw/master/DESCRIPTION", 
                devtools:::github_auth(github_pat()))
```
but it did not worded. So I used, 
```r
httr::GET("https://raw.githubusercontent.com", 
                path = "cderv/myprivaterepo/master/DESCRIPTION", 
                devtools:::github_auth(github_pat()))
```
and it worked

Is it a known issue ? 
Is there a workaround to work with private repo ?

**Comment by cderv** [comment]
Looking for a solution, I find several things : 

- My problem is related to #1280 but it does not seems to work even with `GITHUB_PAT`
- My issue is in fact a duplicate of # 1262 and PR #1263 should resolves it (sorry I did not see it before)
- I found that [`remotes` package](https://github.com/MangoTheCat/remotes) seems to be the new packages for installing functions (#1310) . So I tried it.  

**Conclusion** :  It works fine, provided that on Windows, proxy configuration is correctly done (see MangoTheCat/remotes#45)

**Comment by dy-kim** [comment]
@cderv I've resolved this issue by PR #1382. May be you can find another solution on this.

**Comment by hadley** [comment]
Moving to a meta issue
